test/cassandra/container.spec.ts

```diff
@@ -0,0 +1,93 @@
+import "expect-puppeteer";
+import crypto from 'crypto'
+
+jest.setTimeout(300000);
+
+describe('Collection Add and Delete Cassandra spec', () => {
+  it('creates a collection', async () => {
+    try {
+      const keyspaceId = `keyspaceid${crypto.randomBytes(8).toString("hex")}`;
+      const tableId = `tableid${crypto.randomBytes(3).toString('hex')}`;
+      const prodUrl = "https://localhost:1234/hostedExplorer.html";
+      page.goto(prodUrl);
+
+      // log in with connection string
+      const handle = await page.waitForSelector('iframe');
+      const frame = await handle.contentFrame();
+      await frame.waitFor('div > p.switchConnectTypeText', { visible: true });
+      await frame.click('div > p.switchConnectTypeText');
+      const connStr = process.env.CASSANDRA_CONNECTION_STRING;
+      await frame.type("input[class='inputToken']", connStr);
+      await frame.click("input[value='Connect']");
+
+      // create new table
+      await frame.waitFor('button[data-test="New Table"]', { visible: true });
+      await frame.waitForSelector('div[class="splashScreen"] > div[class="title"]', { visible: true });
+      await frame.click('button[data-test="New Table"]');      
+
+      // type keyspace id
+      await frame.waitFor('input[id="keyspace-id"]', { visible: true });
+      await frame.type('input[id="keyspace-id"]', keyspaceId);      
+
+      // type table id
+      await frame.waitFor('input[class="textfontclr"]');
+      await frame.type('input[class="textfontclr"]', tableId); 
+
+      // click submit
+      await frame.waitFor('#cassandraaddcollectionpane > div > form > div.paneFooter > div > input');
+      await frame.click('#cassandraaddcollectionpane > div > form > div.paneFooter > div > input');
```

Comment:
Is there a way to select that input without relying on the intermediary elements/classes? (which will break if someone changes the layout).

Reply:
Yeah, I think so. It's somewhat difficult to get puppeteer to find these elements without the tree as it doesn't reload the DOM tree during the tests. I was able to remove a lot of these "drilling in" selectors in the SQL spec so will give it a go for Cassandra too